src/scripts/templates.js

```diff
@@ -1,238 +1,192 @@
+    if (isSelectOneElement) div.tabIndex = 0;
+    if (isSelectElement) {
+      div.setAttribute('role', searchEnabled ? 'combobox' : 'listbox');
+      if (searchEnabled) div.setAttribute('aria-autocomplete', 'list');
     }
+    div.setAttribute('aria-haspopup', 'true');
+    div.setAttribute('aria-expanded', 'false');
 
-    return strToEl(`
-      <div
-        class="${globalClasses.containerOuter}"
-        data-type="${passedElementType}"
-        ${role}
-        ${tabIndex}
-        ${ariaAutoComplete}
-        aria-haspopup="true"
-        aria-expanded="false"
-        dir="${direction}"
-        >
-      </div>
-    `);
+    return div;
   },
-  containerInner(globalClasses) {
-    return strToEl(`
-      <div class="${globalClasses.containerInner}"></div>
-    `);
+  containerInner({ containerInner }) {
+    return Object.assign(document.createElement('div'), {
+      className: containerInner,
+    });
   },
-  itemList(globalClasses, isSelectOneElement) {
-    const localClasses = classNames(globalClasses.list, {
-      [globalClasses.listSingle]: isSelectOneElement,
-      [globalClasses.listItems]: !isSelectOneElement,
+  itemList({ list, listSingle, listItems }, isSelectOneElement) {
+    return Object.assign(document.createElement('div'), {
+      className: `${list} ${isSelectOneElement ? listSingle : listItems}`,
     });
-
-    return strToEl(`
-      <div class="${localClasses}"></div>
-    `);
   },
-  placeholder(globalClasses, value) {
-    return strToEl(`
-      <div class="${globalClasses.placeholder}">
-        ${value}
-      </div>
-    `);
+  placeholder({ placeholder }, value) {
+    return Object.assign(document.createElement('div'), {
+      className: placeholder,
+      innerHTML: value,
+    });
   },
-  item(globalClasses, data, removeItemButton) {
-    const ariaSelected = data.active ? 'aria-selected="true"' : '';
-    const ariaDisabled = data.disabled ? 'aria-disabled="true"' : '';
 
-    let localClasses = classNames(globalClasses.item, {
-      [globalClasses.highlightedState]: data.highlighted,
-      [globalClasses.itemSelectable]: !data.highlighted,
-      [globalClasses.placeholder]: data.placeholder,
+  item(
+    { item, button, highlightedState, itemSelectable, placeholder },
+    {
+      id,
+      value,
+      label,
+      customProperties,
+      active,
+      disabled,
+      highlighted,
+      placeholder: isPlaceholder,
+    },
+    removeItemButton,
+  ) {
+    const div = Object.assign(document.createElement('div'), {
+      className: item,
+      innerHTML: label,
     });
+    Object.assign(div.dataset, {
+      item: '',
+      id,
+      value,
+      customProperties,
+    });
+    if (active) div.setAttribute('aria-selected', 'true');
+    if (disabled) div.setAttribute('aria-disabled', 'true');
+
+    if (isPlaceholder) div.classList.add(placeholder);
+    if (highlighted) div.classList.add(highlightedState);
+    else div.classList.add(itemSelectable);
```

Comment:
Can we do normal if/else blocks please :) surprised eslint didn't moan at this

Reply:
Do you mean ‘{}’? Prettier config removing that rule that why no warning here. Will do.